test/test_torch.py

```diff
@@ -3129,6 +3129,10 @@ def test_logical_ops(self):
             else:
                 self.assertFalse(x[idx] ^ y[idx])
 
+        invert_result = ~x
+        for idx in iter_indices(x):
+            self.assertIs(1 - x[idx], invert_result[idx])
```

Comment:
It's not a correct way to test it. It passes only because of an implementation detail of CPython (integers up to 255 are pre-allocated and cached). You should use `self.assertEqual`.